internal/autodetect/main.go

```diff
@@ -182,6 +184,38 @@ func (a *autoDetect) TargetAllocatorAvailability() (targetallocator.Availability
 	return targetallocator.NotAvailable, nil
 }
 
+// CollectorAvailability checks if OpenTelemetryCollector CR are available.
+func (a *autoDetect) CollectorAvailability() (collector.Availability, error) {
```

Comment:
should we merge this with the target allocator availability option and cover the other CRDs too?

Reply:
I don't follow, sorry.

Reply:
should we also do this for the other CRDs (target allocator, instrumentation, opampbridge) so that we don't crash if those aren't present?

Reply:
target allocator already has the logic: https://github.com/open-telemetry/opentelemetry-operator/blob/main/main.go#L389

If you don't mind, I'd leave this for the next PR.